### PR TITLE
add phase 1 plugin support

### PR DESCRIPTION
**Pull request by suz-stripe**
### Reviewers
r? @pepin-stripe 
cc @stripe/developer-products @htheodore-stripe 

 ### Summary
Please see product brief for more details.

### CODE REVIEW

**Comment by Dqddyvillager1722** [comment]
> ### Reviewers
> r? @ cc @stripe/developer-products
> 
> ### Summary
> Please see product brief for more details.

**Review by Dqddyvillager1722 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -73,6 +73,12 @@ build-all-platforms:
 	env GOOS=windows go build -o stripe-windows.exe cmd/stripe/main.go
 .PHONY: build-all-platforms
```
Comment (Dqddyvillager1722) [inline Makefile:74]: @suz-stripe

File: pkg/plugins/plugin.go [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+var (
+	PluginDev   = false
+	PluginsPath string
+)
+
+// Plugin contains the plugin properties
+type Plugin struct {
+	Shortname        string
+	Binary           string
+	Release          []Release
```
Comment (suz-stripe) [inline pkg/plugins/plugin.go:36]: look into whether we can use an annotation to rename to `Releases`

File: pkg/plugins/plugin.go [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+// LookUpLatestVersion gets latest CLI version
+// note: assumes versions are listed in asc order
+func (p *Plugin) LookUpLatestVersion() string {
+	opsystem := runtime.GOOS
+	arch := runtime.GOARCH
+
+	var version string
+	for _, pkg := range p.Release {
+		if pkg.OS == opsystem && pkg.Arch == arch {
```
Comment (suz-stripe) [inline pkg/plugins/plugin.go:145]: explain this iteration getting all the way to the end

**Review by gracegoo-stripe — APPROVED** [review]

**Pull request merged**